File: HackerRank/Staircase/src/Staircase.java
import java.io.*;
import java.util.*;

public class Staircase {
	public static final boolean DEBUG = false;
    public Scanner KB = new Scanner(System.in);
    public static void main(String[] args) {
        if(args.length>0){
			Staircase a = new Staircase(Integer.parseInt(args[0]));
		}
		else{
			Staircase a = new Staircase();
		}
    }
    public Staircase(){
        int input = getInput();
        printStaircaseOfSize(input);
	}
	public Staircase(Integer input){
		printStaircaseOfSize(input);
	}
    public void printStaircaseOfSize(int in){
		for(int row = 0; row < in; row++){
            int col = 0;
            while(col < (in-row-1)){
				//if(DEBUG){System.out.print(" " + col);}
				if(!DEBUG){System.out.print(" ");}
				col ++;
			}
			while(col >= (in - row-1) && col < in){
				//if(DEBUG){System.out.print(" " + col);}
				if(!DEBUG){System.out.print("#");}
				col++;
			}
			System.out.println("");
        }
    }
    public int getInput(){
        int input = KB.nextInt();
        return input;
    }
}

File: HackerRank/Staircase/src/staircase.kt

class Staircase(val size: Int){
    fun build(){
        for (row in 0..size-1){
            for (col in 0..size-1){
                if (col<size-row-1){
                    print(' ')
                } else {
                    print('#')
                }
            }
            println()
        }
    }
}

fun checkStringValidity(str: String?): Boolean {
  if (str is String){
    if (str.length > 0){
      return true
    } else {
      return false
    }
  } else {
    return false
  }
}

fun main(args: Array<String>){
    var size: Int
    var input: String?
    if (args.size == 1) {
      input = args[0]
    } else {
      print("Size of staircase: ")
      input = readLine()
    }
    if (checkStringValidity(input)) {
      size = input!!.toInt()
    } else {
      throw IllegalArgumentException("Input cannot be converted to an int!")
    }
    Staircase(size).build()
}
